lib/doorkeeper/request.rb

```diff
@@ -33,9 +33,17 @@ def token_grant_types
       end
 
       def build_strategy_class(grant_or_request_type)
-        strategy_class_name = grant_or_request_type.to_s.tr(" ", "_").camelize
+        strategy_class_name = if grant_or_request_type.to_s.start_with?(extension_grant_start)
```

Comment:
I need to think more here... :hourglass:

Reply:
I must admit, that I don't like the specs of oauth about extension grant types - but that's not coming from the draft. So I assume, we cannot easily break this down. Else I would like to use grant_type=device_code. On the other hand, maybe this enables others to add extension grant types, like the saml2 bearer in oauth standard grant_type descriptor.
Just for reference:
https://tools.ietf.org/html/draft-ietf-oauth-v2-22#section-4.5